cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/file/FileSetDataset.java

```diff
@@ -175,38 +163,45 @@ public void close() throws IOException {
   @Override
   @SuppressWarnings("unchecked")
   public <T> Class<? extends T> getInputFormatClass() {
-    return (Class<? extends T>) inputFormatClass;
+    return (Class<? extends T>) getFormat(inputFormatClassName, InputFormat.class, classLoader);
   }
 
   @Override
   public Map<String, String> getInputFormatConfiguration() {
-    StringBuilder paths = new StringBuilder();
-    String sep = "";
-    for (Location location : inputLocations) {
-      paths.append(sep).append(getFileSystemPath(location));
-      sep = ",";
-    }
-    return getFormatConfiguration(FileSetProperties.INPUT_PROPERTIES_PREFIX,
-                                  ImmutableMap.of("mapred.input.dir", paths.toString()));
+    String inputs = Joiner.on(',').join(Iterables.transform(inputLocations, new Function<Location, String>() {
+      @Override
+      public String apply(@Nullable Location location) {
+        return getFileSystemPath(location);
+      }
+    }));
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    builder.putAll(FileSetProperties.getInputProperties(properties));
+    builder.putAll(FileSetProperties.getInputProperties(runtimeArguments));
```

Comment:
Will `runtimeArguments` contains the same key as `properties`? Because `ImmutableMap` doesn't allow putting with the same key (it will throw exception).